packages/api/cms-api/src/sentry/sentry.intercepor.ts

```diff
@@ -0,0 +1,74 @@
+import { CallHandler, ExecutionContext, Injectable, NestInterceptor } from "@nestjs/common";
+import { GqlContextType, GqlExecutionContext } from "@nestjs/graphql";
+import { Observable, tap } from "rxjs";
+
+type SentryInterceptorShouldReport = (exception: unknown) => boolean;
+
+export type SentryInterceptorOptions = {
+    shouldReport?: SentryInterceptorShouldReport;
+};
+
+type Sentry = typeof import("@sentry/node");
+
+@Injectable()
+export class SentryInterceptor implements NestInterceptor {
+    shouldReport: SentryInterceptorShouldReport;
+    sentry: Sentry;
+
+    constructor({ shouldReport }: SentryInterceptorOptions, sentry: Sentry) {
+        this.sentry = sentry;
+        this.shouldReport = (error: unknown) => {
+            if (shouldReport) {
+                return shouldReport(error);
```

Comment:
What is the use case for shouldReport being a method in this service? Any why not create a normal class method?

Reply:
I removed the service method. Now it used the function defined in the Project (shouldReportException).
Each project can define when an error should be reported to Sentry.